src/pudl_archiver/archivers/epacems.py

```diff
@@ -66,56 +69,87 @@
+            params=self.parameters,
+            timeout=300,
+        )
+        if file_list.status_code == 200:
+            resjson = file_list.content.decode("utf8").replace("'", '"')
+            file_list.close()  # Close connection.
+            bulk_files = json.loads(resjson)
+            hourly_state_emissions_files = [
+                file
+                for file in bulk_files
+                if (file["metadata"]["dataType"] == "Emissions")
+                and (file["metadata"]["dataSubType"] == "Hourly")
+                and ("stateCode" in file["metadata"].keys())
+            ]
+            logger.info(
+                f"Downloading {len(hourly_state_emissions_files)} total files. This will take more than one hour to respect API rate limits."
+            )
+            if len(hourly_state_emissions_files) > 1000:
+                # This could be used to add pauses to later requests, but in practice
+                # read time is so slow that this takes several hours with concurrency.
+                # If this changes, the request_counter could be used to add a delay.
+                request_counter = 0
+            else:
+                request_counter = np.nan
+            for cems_file in hourly_state_emissions_files:
+                if request_counter is not None:
+                    request_counter += 1
+                if self.valid_year(cems_file["metadata"]["year"]):
+                    yield self.get_state_year_resource(
+                        file=cems_file, request_count=request_counter
+                    )
+        else:
+            raise AssertionError(
+                f"EPACEMS API request did not succeed: {file_list.status_code}"
+            )
 
     async def get_state_year_resource(
-        self, year: int, state: str, months: list[int]
+        self, file: dict[str, str | dict[str, str]], request_count: int | None
     ) -> tuple[Path, dict]:
-        """Download all available months of data for a single state/year."""
-        logger.info(f"Downloading EPACEMS data for {state.capitalize()}, {year}")
-        # Create zipfile to store year/state combinations of files
-        archive_path = self.download_directory / f"epacems-{year}-{state}.zip"
+        """Download all available data for a single state/year.
 
-        for month in months:
-            filename = f"{year}{state}{month:02}.zip"
-            url = f"{BASE_URL}/{year}/{filename}"
-
-            with io.BytesIO() as f_memory:
-                await self.download_zipfile(url, f_memory, ssl=False)
-
-                # Write to zipfile
-                with zipfile.ZipFile(
-                    archive_path, "a", compression=zipfile.ZIP_DEFLATED
-                ) as archive:
-                    with archive.open(filename, "w") as f_disk:
-                        f_disk.write(f_memory.read())
+        Args:
+            file: a dictionary containing file characteristics from the EPA API.
```

Comment:
non-blocking: Is there a published schema we expect this metadata to follow? If so, might be nice to link to it here so we can find it the next time this all breaks down.